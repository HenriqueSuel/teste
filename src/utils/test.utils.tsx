import { useGithub, IGithub } from '@/context/github.context';



export const mockGithubContext = (github?: Partial<IGithub>): void => {
    const mockUseGithub = useGithub as jest.MockedFunction<() => IGithub>;
    mockUseGithub.mockReturnValueOnce(({
        handleGetUser: jest.fn(),
        listUser: [],
        ...github,
    }));
};


