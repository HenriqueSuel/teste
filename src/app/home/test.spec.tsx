// Home.test.jsx
import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import Home from '@/app/home/page'
import { mockGithubContext } from '@/utils/test.utils'; ``

const push = jest.fn();

jest.mock('next/navigation', () => ({
    useRouter: () => ({
        push,
    }),
}));


describe('Home Component', () => {

    beforeEach(() => {
        jest.clearAllMocks();
    })
    it('deve renderizar corretamente', () => {
        mockGithubContext();
        const { container, getByText, getByTestId } = render(<Home />);

        expect(container.firstChild).toMatchSnapshot();
        expect(getByText('Buscar')).toBeTruthy();
        expect(getByTestId('aqui')).toBeTruthy();
    });

    it('deve chamar handleGetUser ao clicar no botão', async () => {
        const handleGetUser = jest.fn();
        mockGithubContext({ handleGetUser })

        const { getByText } = render(<Home />);
        fireEvent.click(getByText('Buscar'));

        await waitFor(() => {
            expect(handleGetUser).toHaveBeenCalled();
        });
    });

    it('deve exibir a lista de usuários', () => {
        const listUser = [
            { id: 1, login: 'user1', name: 'User 1', avatar_url: 'url1' },
            { id: 2, login: 'user2', name: 'User 2', avatar_url: 'url2' },
        ];

        mockGithubContext({ listUser });

        const { getByText } = render(<Home />);

        listUser.forEach((user) => {
            expect(getByText(user.name)).toBeTruthy();
            expect(getByText('Buscar')).toBeTruthy();
        });

        fireEvent.click(getByText(listUser[0].name));

        expect(push).toHaveBeenCalledWith('/repo/user1')
    });
});
