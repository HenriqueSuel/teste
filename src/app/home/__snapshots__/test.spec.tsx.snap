// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component deve renderizar corretamente 1`] = `
<div
  class="container mx-auto"
>
  <div
    class="flex mt-5"
    data-testid="aqui"
  >
    <input
      class="px-2 py-1 border-2 border-blue-300 w-full"
      type="text"
    />
    <button
      class="ml-6 bg-blue-700 px-4 py-2 text-white rounded-lg hover:opacity-90"
    >
      Buscar
    </button>
  </div>
</div>
`;
